contents/data/databases.md

```diff
@@ -0,0 +1,92 @@
+# Introduction to Databases & Database Management Systems (DBMS)
+
+Authors: [Syed Abdullah](https://github.com/Skaty)
+
+## Introduction
+
+![DB-DBMS Relationship](images/db-dbms-relation.png)
+
+A **database** is a collection of data. While any data in a medium can theoretically be a database (for instance, scribbling on a piece of paper), databases in the context of this guide typically involve data that are organised in some manner.
+
+Programs generally do not access the raw data directly. Instead, a  **database management system (DBMS)** is used, which will handle the storage, retrieval and updating of the data.
+
+Sometimes people use the term database to refer to a DBMS. However, to avoid any confusion, this guide will use these terms as defined above.
+
+There are various concepts that are used in the context of databases and DBMS. This guide will cover the basic concepts that are useful for a beginner to start out with.
+
+## Database models
+
+For databases to make any sense, there has to be a certain logical structure in the database (for instance, how data is stored). This logical structure is known as a **database model**.
+
+There are many types of database models. Some examples are:
+
+- **Relational model** - the most widely used database model, usually modeled using a table format. The data structure is defined by a set of relations and domains (types) that dictates the constraints. Constraints are also established for the type of operations that can be done.
+- **Document-oriented** - data is stored in documents, that encapsulates the data. This data is normally stored in a semi-structured manner.
+- **Flat** - data stored as files, without any structure
+
+[Click here](https://en.wikipedia.org/wiki/Database_model) to learn more about the different database models. However, for the purposes of introduction, we would mainly cover on aspects that are used in the relational and/or document-oriented database models.
+
+## Relation
+
+The [proper, formal definition by E. F. Codd](https://en.wikipedia.org/wiki/Relation_(database)) would not be covered here, as it requires the understanding of first-order logic. As such, this section will cover this concept in a simplified manner.
```

Comment:
Can probably put this at the end of the section. Something like "For a more formal definition of a relation, refer to ...". I think it's understood that this is introductory.